src/baseConfigurationProvider.ts

```diff
@@ -0,0 +1,53 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { AnyResolvingConfiguration, ResolvedConfiguration } from './configuration';
+import { fulfillLoggerOptions } from './common/logging';
+
+/**
+ * Base configuration provider that handles some resolution around common
+ * options and handles errors.
+ */
+export abstract class BaseConfigurationProvider<T extends AnyResolvingConfiguration>
```

Comment:
Since logging configuration is shared between all configuration providers, I refactored slightly and created the BaseConfigurationProvider which can handle these common options.